### ISSUE TITLE
Questions related to the finetuning process of Flaubert on a dataset ?

### ISSUE THREAD

**Issue by keloemma** [getalp/Flaubert#11]
I am trying to finetune the model I obtained after training. I some have questions : 

1. firstly, the command below  split the original dataset which is obtained with "get-data-cls.sh" : 

python flue/extract_split_cls.py --indir $DATA_DIR/raw/cls-acl10-unprocessed
--outdir $DATA_DIR/processed
--do_lower true
--use_hugging_face true, 

So when looking at the dataset in the raw directory, I observed that the script " get-data-cls.sh" actually split also the dataset :

![image](https://user-images.githubusercontent.com/40454218/73879609-1766e880-485d-11ea-90a9-6bcd2cbc50a9.png)

2. Since , I already have my own_data, Should I write a new script which will divide my data in order to get the result of the first script  and then use the results as input for the second command "extract_split_cls.py" to obtain the real data to use for finetuning. 

3. Since I have different data , I supposed I should modify the script extract_split_cls.py to use on my own or it is not necessary ?

Output for the second script "extract_split_cls.py"

![image](https://user-images.githubusercontent.com/40454218/73879683-3a919800-485d-11ea-9f79-2471a636c261.png)

4. There is also a "parse.py" script , At which level is it used in the flow of the architecture ? 

![image](https://user-images.githubusercontent.com/40454218/73880393-a9232580-485e-11ea-919f-a9495415111a.png)

5. About the finetuning, what is the utility of the config file ? How is it obtained ? 
=>
config='flue/examples/cls_books_lr5e6_hf_base_uncased.cfg'
source $config

**Comment by formiel** [comment]
Hi @keloemma,

_Question 1, 2, and 3:_ The goal of the script `extract_split_cls.py` is to prepare data in the proper format for the fine-tuning script. In particular, it extracts the text, get the labels, and split the training set into train and dev sets since there is no dev set provided separately for this task. So you should modify or create your own script to process your data if your data is different from the CLS dataset.  

The format of the output files depend on the task. If you use Hugging Face's `transformer` library for finetuning, you can check out the details of the output format [here](https://github.com/huggingface/transformers/blob/master/src/transformers/data/processors/glue.py) if your task is similar to the ones in the GLUE benchmark and prepare your data accordingly.

> There is also a "parse.py" script , At which level is it used in the flow of the architecture ?

It is included in the downloaded data but we do not use it in our code.

> About the finetuning, what is the utility of the config file ? How is it obtained ?

The configuration file is used to save different training parameters. You can run your experiments with different configurations using the same running command. For more details, you can check out the configuration files in the examples and see the parameters. These parameters are used in the running command.

**Comment by schwabdidier** [comment]
I assume that you got your answer @keloemma ?
